config/v1/types_infrastructure.go

```diff
@@ -64,6 +64,7 @@ type InfrastructureStatus struct {
 	// etcdDiscoveryDomain is the domain used to fetch the SRV records for discovering
 	// etcd servers and clients.
 	// For more info: https://github.com/etcd-io/etcd/blob/329be66e8b3f9e2e6af83c123ff89297e49ebd15/Documentation/op-guide/clustering.md#dns-discovery
+	// +optional
```

Comment:
If this is not needed, shouldn't we deprecate the field?

Reply:
I pushed a new commit with the field commented as deprecated.